### ISSUE TITLE
Save and Checkpoint after a fixed time & then resume

### ISSUE THREAD

**Issue by Grant-E-G** [ray-project/ray#9769]
I'm using Tune to train 3 rllib ppo agents in a custom environment. However my cluster has a very low fixed time limit of 6 hours.  I would like to be able to checkpoint after 5Hours and 50mins, and then resume.

My current attempt involved creating a custom stopper for a fixed time, but then I couldn't resume from the saved checkpoint as the job had been Terminated. My next approach would be to edit the json file for the job to remove the Termination flag before resuming, but it would be nice if there was a much cleaner way of doing this. 

Thank you for any help you can provide.

**Comment by richardliaw** [comment]
Hey sorry for the slow response. You probably don't want to use the Stopper to do this. 

Maybe try just running it until it gets killed, and use `tune.run(resume=previous_experiment_dir)`? That should resume from where you left off, if the trials were not stopped.

**Comment by Grant-E-G** [comment]
I tried this before, but my reloads seem to be of checkpoints that are quite dated, and it results in weird plots in tensorboard. 
Likewise the learning rates seem to change after each reload dramatically. Is any of this behavior to be expected/reasonable?

![50dim_episode_reward_mean](https://user-images.githubusercontent.com/126288/89079138-73618400-d34b-11ea-87dc-9f726b3f0cde.png)

**Comment by richardliaw** [comment]
How often are you checkpointing? 

Also, perhaps in your `save/restore` you may want to save the state of the learning rate optimizer, as it seems like the learning rate schedule is being reset after each restore?
